models/mnist_gan_graph.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
+import tensorflow as tf
+import tensorflow.contrib.summary as tf_summary
+
+from models.base import BaseModel
+
+# Flags
+from flags import flags_parser
+FLAGS = flags_parser.FLAGS
+assert FLAGS is not None
+
+
+# Utils
+def next_batch(arr, batch_size):
+    num_batches = int(len(arr) / batch_size)
+    for i in range(0, num_batches * batch_size, batch_size):
+        yield arr[i:i + batch_size]
+    yield arr[num_batches * batch_size:]
+
+
+def noise(size, dist='uniform'):
+    if dist == 'uniform':
+        return np.random.uniform(-1, 1, size=size)
+    elif dist == 'normal':
+        return np.random.normal(size=size)
+    elif dist == 'linspace':
+        n, dim = np.sqrt(size[0]).astype(np.int32), size[1]
+        interpolated_noise = []
+        starts, ends = noise((n, dim)), noise((n, dim))
+        for i in range(n):
+            for w in np.linspace(0, 1, n):
+                interpolated_noise.append(starts[i] + (ends[i] + starts[i]) * w)
+        return np.asarray(interpolated_noise)
+
+
+def shuffle(a, b):
+    # Generate the permutation index array.
+    permutation = np.random.permutation(a.shape[0])
+    # Shuffle the arrays by giving the permutation in the square brackets.
+    return a[permutation], b[permutation]
+
+
+def tile_images(images, num_x, num_y, h, w):
+    res = tf.zeros((num_y * h, num_x * w))
+    index = -1
```

Comment:
use index = i*num_y+j  instead of incrementing index.

Reply:
Not really relevant, I'll add a TODO.